### PR TITLE
[improve][build] Upgrade snakeyaml version to 2.0

### LINKED ISSUE #20013: CVE-2022-1471 identified on pulsar client for snakeyaml's SafeConstructor usage mandate
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

2.10.3

### Minimal reproduce step

Run the security scan 

### What did you expect to see?

Upgrade the snake yaml to latest version 

### What did you see instead?

Vulnerability identified on client modules . 

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation

To avoid [CVE-2022-1471](https://nvd.nist.gov/vuln/detail/CVE-2022-1471)

Jackson needs to upgrade to 2.14.2 and kubenate-client needs to upgrade 18.0.0 to match the snakeyaml version.

When upgrading kubenate-client to 18.0.0, some APIs have changed, so we need to change some related classes.

This closes #20013.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]
Cool!

This should close https://github.com/apache/pulsar/issues/20013.

File: src/owasp-dependency-check-suppressions.xml [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
     <suppress>
         <notes><![CDATA[
-       file name: snakeyaml-1.32.jar
+       file name: snakeyaml-2.0.jar
        ]]></notes>
         <sha1>e80612549feb5c9191c498de628c1aa80693cf0b</sha1>
         <cve>CVE-2022-1471</cve>
```
Comment (lhotari) [inline src/owasp-dependency-check-suppressions.xml:45]: I guess this could be removed instead

**Review by lhotari — APPROVED** [review]

File: src/owasp-dependency-check-suppressions.xml (hunk 1 above)
Comment (Technoboy-) [reply src/owasp-dependency-check-suppressions.xml:45]: Ok

**Comment by tisonkun** [comment]
Shall we cherry-pick to 3.0.0-rc also? cc @poorbarcode @RobertIndie

**Pull request merged**